pkg/river/river.go

```diff
@@ -0,0 +1,322 @@
+// Package river implements a high-level API for decoding and encoding River
+// configuration files. The mapping between River and Go values is described in
+// the documentation for the Unmarshal and Marshal functions.
+//
+// Lower-level APIs which give more control over configuration evaluation are
+// available in the inner packages. The implementation of this package is
+// minimal and serves as a reference for how to consume the lower-level
+// packages.
+package river
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/grafana/agent/pkg/river/parser"
+	"github.com/grafana/agent/pkg/river/token/builder"
+	"github.com/grafana/agent/pkg/river/vm"
+)
+
+// Marshal returns the pretty-printed encoding of v as a River configuration
+// file. v must be a Go struct with river struct tags which determine the
+// structure of the resulting file.
+//
+// Marshal traverses the value v recursively, encoding each struct field as a
+// River block or River attribute, based on the flags provided to the river
+// struct tag.
+//
+// When a struct field represents a River block, Marshal creates a new block
+// and recursively encodes the value as the body of the block. The name of the
+// created block is taken from the name specified by the river struct tag.
+//
+// Struct fields which represent River blocks must be either a Go struct or a
+// slice of Go structs. When the field is a Go struct, its value is encoded as
+// a single block. When the field is a slice of Go structs, a block is created
+// for each element in the slice.
+//
+// When encoding a block, if the inner Go struct has a struct field
+// representing a River block label, the value of that field is used as the
+// label name for the created block. Fields used for River block labels must be
+// the string type. When specified, there must not be more than one struct
+// field which represents a block label.
+//
+// The river tag specifies a name, possibly followed by a comma-separated list
+// of options. The name must be empty if the provided options do not support a
+// name being defined. The following provides examples for all supported struct
+// field tags with their meanings:
+//
+//	// Field appears as a block named "example". It will always appear in the
+//	// resulting encoding. When decoding, "example" is treated as a required
+//	// block and must be present in the source text.
+//	Field struct{...} `river:"example,block"`
```

Comment:
Is there a difference to explain between having `struct {...}` or `*struct{...}` here?

Reply:
Nope, either is fine.